src/reducers/copy.js

```diff
@@ -0,0 +1,127 @@
+      // Koska vain kontrollikentillä on 'value'-stringi suoraan fieldissä, datakentillä joka subfieldin alla
+      if ('value' in sourceField) {
+        // Jos isIdentical palauttaa falsen, ei löydy matchia eli ei ole identtisiä kontrollikenttiä
+        // (eli koko if-blokin arvo on true)
+        debug('tarkistetaan kontrollikentät');
+        debug(`sourceField kontrollikentät on: ${JSON.stringify(sourceField, undefined, 2)}`);
+        return baseFields.some(isIdenticalControlField) === false;
+      }
+
+      // Jos sourceFieldillä on subfields-osio, kyseessä on datakenttä.
+      // Testi 04: Onko identtisiä datakenttiä? (04:ssä on 2 kpl)
+      // Testi 05: Yksi samanlainen ja yksi erilainen datakenttä
+      if ('subfields' in sourceField) {
+        debug('tarkistetaan datakentät');
+        debug(`sourceField datakentät on: ${JSON.stringify(sourceField, undefined, 2)}`);
+        return baseFields.some(isIdenticalDataField) === false;
+      }
+
+      // Debug(`filterMissingin arvo on: ${JSON.stringify(filterMissing, undefined, 2)}`);
+      // Return filterMissing; // Palauttaa ei-identtiset kentät
+
+      // Apufunktiot:
+      // Marc-kentän normalisointi
+      function normalizeControlField(field) {
+        return field.value.toLowerCase().replace(/\s+/u, '');
+      }
+      function normalizeDataField(field) {
+        // Return field.subfields[0].value.toLowerCase().replace(/\s+/u, ''); // tämä toimii mutta vain ekalle indeksille
+        const x = field.subfields.map(subfield => subfield.value.toLowerCase().replace(/\s+/u, '')); // Miksi tämä ei toimi?
+        debug(`tulos on ${x}`);
+        return x;
+      }
+
+      // Ovatko normalisoitu base ja source identtisiä?
+      // Jos funktio palauttaa true niin ovat, jos false niin eivät
+      function isIdenticalControlField(baseField) {
+        const normalizedBaseField = normalizeControlField(baseField);
+        const normalizedSourceField = normalizeControlField(sourceField);
+        return normalizedSourceField === normalizedBaseField;
+      }
+
+      // Ovatko datakentät identtisiä? tag, ind1, ind2 ja kaikki osakentät
+      function isIdenticalDataField(baseField) {
+        const normalizedBaseField = normalizeDataField(baseField);
+        const normalizedSourceField = normalizeDataField(sourceField);
+        if (sourceField.tag === baseField.tag &&
+          sourceField.ind1 === baseField.ind1 &&
+          sourceField.ind2 === baseField.ind2 &&
+          sourceField.subfields.length === baseField.subfields.length) {
+        // Jos mikään näistä ei toteudu, palauttaa false
+        // Ja tämän sisään tulee subfieldien vertailu
+        // SourceField.subfields[0].code === baseField.subfields[0].code &&
+        // SourceField.subfields[0].value === baseField.subfields[0].value) {
+          debug('tarkistetaan datakenttien sisällöt');
+          return normalizedSourceField === normalizedBaseField;
```

Comment:
Data fields don't have a single value you can compare. The specification (In Kupoli) says: ` Each subfield must find an identical pair in the another record. The code and value must match with strict string equality. The order of subfields does not matter.`

Since at this point we already know that the number of subfields is identical you can use either either as the iterator: `baseField.subfields.every()`